pkg/ipam/allocator/podcidr/podcidr.go

```diff
@@ -849,12 +851,37 @@ func (n *NodesPodCIDRManager) allocateNext(nodeName string) (*nodeCIDRs, bool, e
 		nCIDRs.v6PodCIDRs = []*net.IPNet{v6CIDR}
 	}
 
+	if nCIDRs == nil {
+		err = fmt.Errorf("Unable to allocate node CIDR for node %s. IPAMInfo: {IPv4: %s, IPv6: %s}. Please check that your configuration is correct",
+			nodeName,
+			getCIDRAllocatorsInfo(n.v4CIDRAllocators, v4AllocatorType),
+			getCIDRAllocatorsInfo(n.v6CIDRAllocators, v6AllocatorType))
+		return nil, false, err
+	}
+
 	n.nodes[nodeName] = nCIDRs
 
 	return nCIDRs, true, nil
 
 }
 
+func getCIDRAllocatorsInfo(cidrAllocators []CIDRAllocator, netTypes string) string {
+	var cidrAllocatorsInfo bytes.Buffer
+	cidrAllocatorsLength := len(cidrAllocators)
+	if cidrAllocatorsLength == 0 {
+		return "[]"
+	}
+
+	for index, cidrAllocator := range cidrAllocators {
+		cidrAllocatorsInfo.WriteString(fmt.Sprintf("%s", cidrAllocator.String()))
+		if index < cidrAllocatorsLength - 1 {
```

Comment:
Please format your code with `gofmt -s` to pass the style checks. In that case, I think it simply expects no space around the minus sign:
```suggestion
		if index < cidrAllocatorsLength-1 {
```